compiler/testData/diagnostics/tests/controlFlowAnalysis/doWhileAssignment.fir.kt

```diff
@@ -3,21 +3,21 @@
 fun test() {
     do {
         val s: String
-        s = ""
+        <!VAL_REASSIGNMENT!>s<!> = ""
```

Comment:
Due to the loop ending condition below (`s == ""`), this shouldn't be the case, but no idea yet where to look at, how to integrate value assignment to CFG building/pruning, etc.

Reply:
This diagnostic is definitely wrong and it prevents this PR from being accepted.

Reply:
It's gone! (And all others!)